lock/lock_windows.go

```diff
@@ -54,12 +54,12 @@ func (u *winUnlocker) close() {
 func lockWindows(name string) (io.Closer, error) {
 	fi, err := os.Stat(name)
 	if err == nil && fi.Size() > 0 {
-		return nil, fmt.Errorf("can't lock file %q: %s", name, "has non-zero size")
+		return nil, fmt.Errorf("can't lock file %s: has non-zero size", name)
```

Comment:
The use of `%q` here was likely intentional.

Using a simple `%s` for filenames can often lead to confusing error output: (https://play.golang.org/p/fiuJEvghF7y)
```
open : no such file or directory
```

While we can clearly see exactly why the output here is what it is in this simple case, in the real world, it can often be much more difficult to realize that there is in fact a filename in that error message.

Reply:
It's true! I put that `%q` there a while ago for that reason. However, when looking at it today I noticed the Unix files use `%s`.
The other issue though, is that while `%q` avoids the empty string issue, it formats to "string safely escaped with Go syntax
".
In this case, it causes "fencing" in Windows paths. We can avoid it though by manually providing quotations here:
https://play.golang.org/p/RO0HBOIrdz7
```suggestion
		return nil, fmt.Errorf(`can't lock file "%s": has non-zero size`, name)
```

The same could be added in the `lock_unix` file, or their `%s` changed to a `%q`.

Which do you think would make more sense to use in the `_unix` file? I'm thinking manual quotes just for consistency.

Reply:
Ah yeah. Hm. An interesting dilemma there. I‘ve simply accepted that backslashes get doubled in the Windows code. I suppose I was already deadened to it after years of dealing with backslash doubling, and quadrupling, and worse while at Microsoft. :laughing:

I’m not sure we want to use manual quotes either though, because then it doesn’t escape quotes in the string as well. Which, sure they cannot appear in Windows filenames, but this is an error message about “cannot make such a file”, so it could still happen.

I mean, also https://play.golang.org/p/pZmjyiqj5M3 this wonderful piece of bad, but I think that one is playing way way more into the weeds of what to protect against. :thinking: Maybe we could just `if name == "" { name = `""` }` :laughing: 

Not sure, I’m down with going to `%s`, the rest of the errors do this, and so do the internal Go libraries. At least we won’t end up doing anything worse than them.